vllm/core/scheduler.py

```diff
@@ -273,11 +274,19 @@ def __init__(
         BlockSpaceManagerImpl = BlockSpaceManager.get_block_space_manager_class(
             version)
 
+        num_gpu_blocks = cache_config.num_gpu_blocks
+        if num_gpu_blocks:
+            num_gpu_blocks //= pipeline_parallel_size
+
+        num_cpu_blocks = cache_config.num_cpu_blocks
+        if num_cpu_blocks:
+            num_cpu_blocks //= pipeline_parallel_size
```

Comment:
what if they are not dividable? Do we need to warn users?

Reply:
It should be ok as we use `//=` so we implicitly use `floor`. We lose max `pipeline_parallel_size - 1` blocks which should be quite small. Let me know if you think this is ok.